CorsixTH/Src/th_gfx.cpp

```diff
@@ -703,13 +703,36 @@ size_t THAnimationManager::getFrameCount() const
 }
 
 size_t THAnimationManager::getFirstFrame(size_t iAnimation) const
+{
+    size_t result = getCustomFirstFrame(iAnimation);
+    if (result > 0)
+        return result;
+
+    return getOriginalFirstFrame(iAnimation);
+}
+
+size_t THAnimationManager::getOriginalFirstFrame(size_t iAnimation) const
 {
     if(iAnimation < m_iAnimationCount)
         return m_vFirstFrames[iAnimation];
     else
         return 0;
 }
 
+size_t THAnimationManager::getCustomFirstFrame(size_t iAnimation) const
+{
+    if (iAnimation > 99999) // Number too large to fit, requested animation doesn't exist.
```

Comment:
Too large to fit where? Could be a constant perhaps?

Reply:
The number is (or rather, was) converted to a string in a fixed size char buffer (so all animations have a name rather than a number), so this is an overflow protection. As this is used only for the original animations, the upper limit is known.
TheCycoONE didn't like that though and wanted a std::string for the buffer and fixed it I think. I have no problems with std::string fixing or with introducing a constant as you suggested, but as this patch isn't anywhere near covering the problem it's trying to address, I didn't fix it since I consider this a failed attempt (but a very successful one for learning about the true problem).